ocrd_tesserocr/deskew.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import absolute_import
+
+import locale
+
+# pylint: disable=wrong-import-position
+locale.setlocale(locale.LC_ALL, 'C') # circumvent tesseract-ocr issue 1670 (which cannot be done on command line because Click requires an UTF-8 locale in Python 3)
+
+from tesserocr import RIL, PSM, PyTessBaseAPI
+
+from ocrd_utils import getLogger, concat_padded, xywh_from_points, points_from_x0y0x1y1, MIMETYPE_PAGE
+from ocrd_modelfactory import page_from_file
+from ocrd_models.ocrd_page import (
+    CoordsType,
+    GlyphType,
+    LabelType,
+    LabelsType,
+    MetadataItemType,
+    TextEquivType,
+    TextStyleType,
+
+    to_xml
+)
+from ocrd import Processor
+from .config import TESSDATA_PREFIX, OCRD_TOOL
+
+log = getLogger('processor.TesserocrDeskew')
+
+class TesserocrDeskew(Processor):
+
+    def __init__(self, *args, **kwargs):
+        kwargs['ocrd_tool'] = OCRD_TOOL['tools']['ocrd-tesserocr-deskew']
+        kwargs['version'] = OCRD_TOOL['version']
+        super(TesserocrDeskew, self).__init__(*args, **kwargs)
+
+    def process(self):
+        """
+        Performs the deskewing.
+        """
+        # print(self.parameter)
+        oplevel = self.parameter['operation_level']
+        with PyTessBaseAPI(path=TESSDATA_PREFIX, psm=PSM.AUTO_OSD) as tessapi:
+            for (n, input_file) in enumerate(self.input_files):
+                log.info("INPUT FILE %i / %s", n, input_file)
+                pcgts = page_from_file(self.workspace.download_file(input_file))
+                pil_image = self.workspace.resolve_image_as_pil(pcgts.get_Page().imageFilename)
+
+                metadata = pcgts.get_Metadata() # ensured by from_file()
+                metadata.add_MetadataItem(
```

Comment:
I copied the `metadata` code from https://github.com/OCR-D/ocrd_tesserocr/blob/d968abcd62dab1f862faa7df4e1676d70322402f/ocrd_tesserocr/recognize.py#L106 ff. No idea what it actually does or if it is necessary. In addition, `segment_region` does not have it.